### PR TITLE
Add `pbxproj_partials.write_bazel_dependencies`

### PR DESCRIPTION
**Pull request by brentleyjones**

### CODE REVIEW

**Comment by brentleyjones** [comment]
I'll rebase on `main` once CI lets me merge the dependent PR. This is ready for review now.

File: test/internal/pbxproj_partials/write_bazel_dependencies_tests.bzl [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+
+    def _actions_write(write_output, args):
+        writes[write_output] = args
+
+    actions = struct(
+        args = lambda: action_args,
+        declare_file = _actions_declare_file,
+        run = _action_run,
+        write = _actions_write,
+    )
```
Comment (brentleyjones) [inline test/internal/pbxproj_partials/write_bazel_dependencies_tests.bzl:66]: After one more set of tests that duplicate this, I'll feel good about pulling it out into a test module that can be reused.

**Review by mattrobmattrob — APPROVED** [review]

**Review by thiagohmcruz — APPROVED** [review]
